includes/class-wc-stripe-intent-controller.php

```diff
@@ -254,6 +256,65 @@ public function create_setup_intent() {
 		echo wp_json_encode( $response );
 		exit;
 	}
+
+	/**
+	 * Handle AJAX request for creating a payment intent for Stripe UPE.
+	 *
+	 * @throws Process_Payment_Exception - If nonce or setup intent is invalid.
+	 */
+	public function create_payment_intent_ajax() {
+		try {
+			$is_nonce_valid = check_ajax_referer( '_wc_stripe_nonce', false, false );
+			if ( ! $is_nonce_valid ) {
+				throw new Exception( __( "We're not able to process this payment. Please refresh the page and try again.", 'woocommerce-gateway-stripe' ) );
+			}
+
+			// If paying from order, we need to get the total from the order instead of the cart.
+			$order_id = isset( $_POST['stripe_order_id'] ) ? absint( $_POST['stripe_order_id'] ) : null;
+
+			wp_send_json_success( $this->create_payment_intent( $order_id ), 200 );
+		} catch ( Exception $e ) {
+			// Send back error so it can be displayed to the customer.
+			wp_send_json_error(
+				[
+					'error' => [
+						'message' => $e->getMessage(),
+					],
+				]
+			);
+		}
+	}
+
+	/**
+	 * Creates payment intent using current cart or order and store details.
+	 *
+	 * @param {int} $order_id The id of the order if intent created from Order.
+	 *
+	 * @return array
+	 */
+	public function create_payment_intent( $order_id = null ) {
+		$amount = WC()->cart->get_total( false );
+		$order  = wc_get_order( $order_id );
+		if ( is_a( $order, 'WC_Order' ) ) {
+			$amount = $order->get_total();
+		}
```

Comment:
Are there any implications here if `$order` is a subclass of `WC_Order`, or do we want `WC_Order`s here specifically? I'm wondering if it'd be better to check if the `$order` variable has a `get_total()` function or if it the `WC_Order` check is preferred 🤔 

For clarity, this is what I mean:

```php
if ( method_exists( $order, 'get_total' ) ) {
```

See [method_exists()](https://www.php.net/manual/en/function.method-exists.php) documentation.

Reply:
Using `method_exists( $order, 'get_total' )` makes sense.. changed.